ra/ra.go

```diff
@@ -1546,10 +1643,41 @@ func (ra *RegistrationAuthorityImpl) PerformValidation(
 		}
 		authz.Challenges[challIndex] = *challenge
 
-		err = ra.onValidationUpdate(vaCtx, authz)
-		if err != nil {
-			ra.log.AuditErrf("Could not record updated validation: err=[%s] regID=[%d] authzID=[%s]",
-				err, authz.RegistrationID, authz.ID)
+		if authz.V2 {
```

Comment:
I think this code would fit better inside `onValidationUpdate`, since it more closely mirrors the code that's already there. That would also reduce the number of nested `if`'s in `PerformValidation` and its overall length.